pan3d/ui/file_select.py

```diff
@@ -0,0 +1,102 @@
+                        click=submit_import,
+                        style="width: 100%",
+                    )
+
+                    vuetify.VTextField(
+                        v_model=selected_config_file,
+                        v_show=f"{dialog_shown} === 'Export' && {selected_config_file} != false",
+                        label="Download Location",
+                        prepend_icon="mdi-paperclip",
+                        # FileSystem API is only available on some browsers:
+                        # https://caniuse.com/native-filesystem-api
+                        click="""
+                            if ($event){
+                                try {
+                                    window.showSaveFilePicker({
+                                        suggestedName: 'pan3d_state.json',
+                                        types: [{
+                                            accept: {
+                                                'application/json': ['.json']
+                                            }
+                                        }]
+                                    }).then((handle) => {
+                                        handle.createWritable().then((writable) => {
+                                            writable.write(JSON.stringify(%s, null, 4));
+                                            writable.close();
+                                            %s = 'Export complete.'
+                                        })
+                                    });
+                                } catch {
+                                    %s = false;
+                                    window.alert('Your browser does not support selecting a download location. Your download will be made to the default location, saved as pan3d_state.json.')
+                                }
+                            }
+                        """
+                        % (state_export, dialog_message, selected_config_file),
+                    )
+                    vuetify.VBtn(
+                        v_show=f"{dialog_shown} === 'Export' && {selected_config_file} === false",
+                        variant="tonal",
+                        text="Download pan3d_state.json",
+                        click="""
+                            var content = JSON.stringify(%s, null, 4);
+                            var a = window.document.createElement('a');
+                            a.href = 'data:application/json;charset=utf-8,'  + encodeURIComponent(content);
+                            a.download = 'pan3d_state.json';
+                            a.style.display = 'none';
+                            window.document.body.appendChild(a);
+                            a.click()
+                            window.document.body.removeChild(a);
+                            %s = 'Export complete.'
+                        """
+                        % (state_export, dialog_message),
```

Comment:
f-string

Reply:
Changed in [28561cb](https://github.com/Kitware/pan3d/pull/33/commits/28561cb331878bed9f5131df5a01d8307360c41e)